onnxruntime/test/providers/qnn/qnn_basic_test.cc

```diff
@@ -1098,6 +1099,38 @@ TEST_F(QnnHTPBackendTests, EPOffloadsGraphIOQuantDequant) {
   }
 }
 
+TEST_F(QnnHTPBackendTests, UseHtpSharedMemoryAllocatorForInputs) {
```

Comment:
We should also have some codes to demonstrate how this feature get used from user code.
Here are some IObinding examples for other EPs:
https://github.com/microsoft/onnxruntime/blob/3b1a9002f507103229f1613193c5647a203d567d/onnxruntime/test/shared_lib/test_inference.cc#L2076

Reply:
added some similar tests using the htp shared memory allocator